### PR TITLE
Support data lake storage gen2 filesystems

### LINKED ISSUE #387: Azure Data Lake Storage Gen2  Filesystem - PQ4
Resource Support for Azure Data Lake Storage Gen2 Filesystem
1.Dev
2.Test
3.Documentation

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description

Support ADLS Gen2 Filesystem(s)

### Issues Resolved

#387 

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/azure_data_lake_storage_gen2_filesystems.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+describe azure_data_lake_storage_gen2_filesystems(account_name: 'ACCOUNT_NAME') do
+  #...
+end
+```
+
+## Parameters
+| Name           | Description                                                                      |
+|----------------|----------------------------------------------------------------------------------|
+| account_name   | The Azure Storage account name.                                                  |
+| dns_suffix     | The DNS suffix for the Azure Data Lake Storage endpoint.                         |
```
Comment (sa-progress) [inline docs/resources/azure_data_lake_storage_gen2_filesystems.md:37]: Can you just confirm .If user want to provide dns_suffix ,How he will do that ?

Comment (sathish-progress) [reply docs/resources/azure_data_lake_storage_gen2_filesystems.md:37]: yeah so the `dns_suffix` could also be a custom suffix, that is why we need provide an optional way to accept it

**Review by sa-progress — APPROVED** [review]
LGTM

File: docs/resources/azure_data_lake_storage_gen2_filesystem.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+Also, refer to [Azure documentation](https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/filesystem/get-properties) for other properties available.
+
+## Examples
+
+### Test that the Data Lake Storage Gen 2 Filesystem 
+
+```ruby
+describe azure_data_lake_storage_gen2_filesystem(account_name: 'ACCOUNT_NAME', name: 'FILE_SYSTEM')  do
+  its() { should eq 1.0 }
```
Comment (IanMadd) [inline docs/resources/azure_data_lake_storage_gen2_filesystem.md:76]: Is this line correct?

Comment (sathish-progress) [reply docs/resources/azure_data_lake_storage_gen2_filesystem.md]: no that is wrong .. fixed it

**Review by IanMadd — APPROVED** [review]

**Pull request merged**